components/storage/blockdevice/COMPONENT_FLASHIAP/TESTS/filesystem/fopen/fopen.cpp

```diff
@@ -78,7 +78,11 @@ static char fslittle_fopen_utest_msg_g[FSLITTLE_UTEST_MSG_BUF_SIZE];
 #define FSLITTLE_FOPEN_TEST_MOUNT_PT_PATH      "/"FSLITTLE_FOPEN_TEST_MOUNT_PT_NAME
 #define FSLITTLE_FOPEN_TEST_WORK_BUF_SIZE_1    64
 #define FSLITTLE_FOPEN_TEST_FILEPATH_MAX_DEPTH 20
+#ifdef TARGET_SAML21J18A
+static const int MAX_TEST_SIZE = 64 * 1024;
```

Comment:
test should not contain specific target. Wasn't this already addressed on master? 

cc @ARMmbed/mbed-os-storage

Reply:
@janjongboom Sorry for missing this earlier. Your target is not the only one with small RAM. I guess you're trying to run all the tests and see if they pass on your target. However, in the LittleFS specific case, tests are only limited in our CI to one or two boards that are capable of running the tests (have enough memory, have persistent storage etc.). So as @0xc0170 said - no target specific ifdefs should appear on the test code.

Reply:
Nevertheless, adding specific target based ifdefs in our test code is not the way to go. Other targets may have less memory (there are such I know), meaning that supporting them would require additional ifdefs. There are dozens of supported targets and choosing this path will make the code unreadable. We can try considering the available memory to determine the test size in this test, but it should be outside this PR's scope IMO.

Reply:
Can we remove this from this PR, create an issue and address this ? If this is causing an issue for this target, would lead to disabling flashiap until the fix is in place - it would not block the further progress.

Reply:
@0xc0170 @davidsaada @cmonr I've moved this to mbed_lib, just like the other flashiap configuration (like where the flash starts). Does this work for you? This removes any ifdef code and the default behavior stays the same.